proxy/src/main/java/com/velocitypowered/proxy/connection/backend/VelocityServerConnection.java

```diff
@@ -140,13 +152,26 @@ String getPlayerRemoteAddressAsString() {
     }
   }
 
+  private String getRemoteAddress() {
+    var op = proxyPlayer.getVirtualHost();
+    if (op.isPresent()) {
+      return op.get().getHostString();
+    }
+    SocketAddress address = registeredServer.getServerInfo().getSocketAddress();
+    if (address instanceof InetSocketAddress) {
+      InetSocketAddress address1 = (InetSocketAddress) address;
+      return address1.getHostString();
+    } else {
+      DomainSocketAddress address1 = (DomainSocketAddress) address;
+      return address1.toString();
+    }
+  }
+
   private String createLegacyForwardingAddress(UnaryOperator<List<Property>> propertiesTransform) {
     // BungeeCord IP forwarding is simply a special injection after the "address" in the handshake,
     // separated by \0 (the null byte). In order, you send the original host, the player's IP, their
     // UUID (undashed), and if you are in online-mode, their login properties (from Mojang).
-    StringBuilder data = new StringBuilder().append(proxyPlayer.getVirtualHost().orElseGet(() ->
-                    registeredServer.getServerInfo().getAddress()).getHostString())
-        .append('\0')
+    StringBuilder data = new StringBuilder().append(getRemoteAddress())
```

Comment:
this changes the intended behavior here

Reply:
ok

Reply:
unix socket hoststring like `/xxx/xxx` is a filename, but inetsocketaddress.getHostString is a ipaddress, because of these two differences, resulting in different behaviors?

Reply:
oh i missing a `.append('\0')`